scripts/provision_health_checks.py

```diff
@@ -0,0 +1,92 @@
+from itertools import chain
+
+import argparse
+import boto3
+import logging
+import json
+import sys
+
+import more_itertools
+
+from azul import config
+from azul.logging import configure_script_logging
+
+log = logging.getLogger(__name__)
+
+
+class ProvisionHealthChecks:
+
+    def __init__(self):
+        self.client = boto3.client('route53')
+
+    @property
+    def get_health_checks(self):
+        paginator = self.client.get_paginator('list_health_checks')
+        health_checks = chain.from_iterable(page['HealthChecks'] for page in paginator.paginate())
+        health_check_batches = more_itertools.chunked(health_checks, 10)
+        dcp_health_checks = {}
+        for health_check_batch in health_check_batches:
+            health_check_batch = {health_check['Id']: health_check for health_check in health_check_batch}
+            response = self.client.list_tags_for_resources(ResourceType='healthcheck',
+                                                           ResourceIds=list(health_check_batch.keys()))
+            for tag_sets in response['ResourceTagSets']:
+                assert tag_sets['ResourceType'] == 'healthcheck'
+                for tag in tag_sets['Tags']:
+                    if tag['Key'] == 'Name':
+                        health_check_name = tag['Value']
+                        health_check_id = tag_sets['ResourceId']
+                        dcp_health_checks[health_check_name] = health_check_batch[health_check_id]
+        return dcp_health_checks
+
+    def provision_health_check(self, health_checks, dcp_health_check, reference):
+        dcp_health_check_children = dcp_health_check['HealthCheckConfig']['ChildHealthChecks']
+        azul_health_checks = []
+        for resource_name in (config.service_name,
+                              config.indexer_name,
+                              config.data_browser_name,
+                              config.data_portal_name):
+            try:
+                azul_health_checks.append(health_checks[resource_name]['Id'])
```

Comment:
```
azul_health_checks.append(health_checks[resource_name]['Id'])
                  ^      ^             ^              ^
```

are the places where technically a KeyError could occur. Only one should be caught and handled in this way. The other ones should cause the program to exit with a Traceback and a non-zero status code (the default behavior for unhandled exceptions).